RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractZoneIdentifier.java

```diff
@@ -0,0 +1,398 @@
+                LOG.log(Level.WARNING, message, ex);
+            }
+        }
+
+        IngestServices services = IngestServices.getInstance();
+
+        if (!sourceArtifacts.isEmpty()) {
+            services.fireModuleDataEvent(new ModuleDataEvent(
+                    RecentActivityExtracterModuleFactory.getModuleName(),
+                    BlackboardArtifact.ARTIFACT_TYPE.TSK_DOWNLOAD_SOURCE, sourceArtifacts));
+        }
+
+        if (!downloadArtifacts.isEmpty()) {
+            services.fireModuleDataEvent(new ModuleDataEvent(
+                    RecentActivityExtracterModuleFactory.getModuleName(),
+                    BlackboardArtifact.ARTIFACT_TYPE.TSK_WEB_DOWNLOAD, downloadArtifacts));
+        }
+    }
+
+    /**
+     * Process a single Zone Identifier file.
+     *
+     * @param context           IngetJobContext
+     * @param dataSource        Content
+     * @param zoneFile          Zone Indentifier file
+     * @param sourceArtifacts   List for TSK_DOWNLOAD_SOURCE artifacts
+     * @param downloadArtifacts List for TSK_WEB_DOWNLOAD aritfacts
+     */
+    private void processZoneFile(IngestJobContext context, Content dataSource,
+            AbstractFile zoneFile, Collection<BlackboardArtifact> sourceArtifacts,
+            Collection<BlackboardArtifact> downloadArtifacts) throws TskCoreException {
+
+        ZoneIdentifierInfo zoneInfo = null;
+
+        try {
+            zoneInfo = new ZoneIdentifierInfo(zoneFile);
+        } catch (IOException ex) {
+            String message = String.format("Unable to parse temporary File for %s", zoneFile.getName()); //NON-NLS
+            LOG.log(Level.WARNING, message, ex);
+        }
+
+        if (zoneInfo == null) {
+            return;
+        }
+
+        AbstractFile downloadFile = getDownloadFile(dataSource, zoneFile);
+        ArrayList<Long> knownPathIDs = getPathIDsForType(BlackboardArtifact.ARTIFACT_TYPE.TSK_WEB_DOWNLOAD);
+
+        if (downloadFile != null) {
+            if (!knownPathIDs.contains(downloadFile.getDataSourceObjectId())) {
+                BlackboardArtifact downloadbba = createDownloadArtifact(zoneFile, zoneInfo);
+                if (downloadbba != null) {
+                    downloadArtifacts.add(downloadbba);
+                }
+            }
+
+            if (downloadFile.getArtifactsCount(BlackboardArtifact.ARTIFACT_TYPE.TSK_DOWNLOAD_SOURCE) == 0) {
+                BlackboardArtifact sourcebba = createDownloadSourceArtifact(downloadFile, zoneInfo);
+                if (sourcebba != null) {
+                    sourceArtifacts.add(sourcebba);
+                }
+            }
+        }
+    }
+
+    /**
+     * Find a list of all the file that end in <i>:Zone.Identifier<i>.
+     *
+     * @param dataSource Content
+     *
+     * @return A list of zone identifier files
+     *
+     * @throws TskCoreException
+     */
+    private List<AbstractFile> findZoneFiles(Content dataSource) throws TskCoreException {
```

Comment:
Personally, I would inline this method, since it is only used in one place and can be a one liner